hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/ReflectiveCompactSerializer.java

```diff
@@ -68,6 +70,17 @@
 import static com.hazelcast.nio.serialization.FieldType.UTF_ARRAY;
 import static java.util.stream.Collectors.toList;
 
+/**
+ * Reflective serializer works for Compact format in zero-config case.
+ * Specifically when explicit serializer is not given via
+ * {@link com.hazelcast.config.CompactSerializationConfig#register(Class, String, CompactSerializer)}
+ * {@link com.hazelcast.config.CompactSerializationConfig#register(Class, CompactSerializer)}
+ * <p>
+ * ReflectiveCompactSerializer can de/serialize classes having an accessible empty constructor only.
+ * Only types in {@link CompactWriter}/{@link CompactReader} interface are supported as fields.
+ * For any other class as the field type, it will work recursively and try to de/serialize a sub-class.
+ * Thus, if any sub-fields does not have an accessible empty constructor, deserialization fails with IOException.
```

Comment:
is it supposed to be `IOException` or `HazelcastSerializationException`?

Reply:
Changing it with `HazelcastSerializationException`